lib/ansible/plugins/lookup/onepassword.py

```diff
@@ -0,0 +1,108 @@
+# (c) 2018, Scott Buchanan <sbuchanan@ri.pn>
+# (c) 2016, Andrew Zenk <azenk@umn.edu> (lastpass.py used as starting point)
+# (c) 2017 Ansible Project
+# GNU General Public License v3.0+ (see COPYING or https://www.gnu.org/licenses/gpl-3.0.txt)
+from __future__ import (absolute_import, division, print_function)
+__metaclass__ = type
+
+ANSIBLE_METADATA = {'metadata_version': '1.1',
+                    'status': ['preview'],
+                    'supported_by': 'community'}
+
+DOCUMENTATION = """
+    lookup: onepassword
+    author:
+      -  Scott Buchanan <sbuchanan@ri.pn>
+      -  Andrew Zenk <azenk@umn.edu>
+    version_added: "TBD"
+    requirements:
+      - jq (python library: https://pypi.python.org/pypi/jq)
+      - op (1Password command line utility: https://support.1password.com/command-line/)
+      - must have already logged into 1Password using op CLI
+    short_description: fetch data from 1Password
+    description:
+      - use the op command line utility to fetch specific fields from 1Password
+    options:
+      _terms:
+        description: identifier(s) of object(s) from which you want to retrieve the field (UUID, name or domain)
+        required: True
+      field:
+        description: field to return from 1Password
+        default: 'password'
+      vault:
+        description: vault to retrieve from (if absent will search all available vaults)
+        default: None
+"""
+
+EXAMPLES = """
+- name: "retrieve password from entry named KITT in any vault"
+  debug: password="{{ lookup('onepassword', 'KITT') }}"
+
+- name: "retrieve username from entry named HAL 9000 in vault Discovery"
+  debug: password="{{ lookup('onepassword', 'HAL 9000', field='username', vault='Discovery') }}"
+"""
+
+RETURN = """
+  _raw:
+    description: field data requested
+"""
+
+from jq import jq
+from subprocess import Popen, PIPE
+
+from ansible.errors import AnsibleError
+from ansible.plugins.lookup import LookupBase
+
+
+class OnePassException(AnsibleError):
+
+    pass
+
+
+class OnePass(object):
+
+    def __init__(self, path='op'):
+        self._cli_path = path
+
+    @property
+    def cli_path(self):
+        return self._cli_path
+
+    def assert_logged_in(self):
+        try:
+            self._run(["get", "account"])
+        except:
+            raise OnePassException("Not logged into 1Password: please run 'op signin' first")
+
+    def get_field(self, item_id, field, vault=None):
+        escaped_field = field.replace('\\', '\\\\').replace('"', '\\"')
+        parser = '.details.sections[].fields[]? | select(.t=="{0}").v'.format(escaped_field)
```

Comment:
I'm curious why the `jq` library was used instead of the native JSON package? (I know we suggest the JQ CLI in our docs, but I imagine with a more capable language like Python, that requirement shouldn't be needed)

Reply:
Oh well it looks like I jumped the gun here, someone already asked that! Disregard, it seems you're on top of it :)